### PR TITLE
Added a BC validation for changing incident field type

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/22016

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 1]
```diff
@@ -158,15 +158,15 @@ class IncidentFieldValidator(BaseValidator):
     """
 
     def is_backward_compatible(self):
-        """Check whether the Incident Field is backward compatible or not, update the _is_valid field to determine that
-        TODO
+        """Check whether the Incident Field is backward compatible or not,
+        update the _is_valid field to determine that
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/incident_field.py]: you are not updating it

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 2]
```diff
@@ -332,9 +332,25 @@ def is_valid_required(self):
...
         if not is_valid:
             print_error(error_msg)
         return is_valid
+
+    def is_changed_type(self):
+        # type: () -> bool
+        """Validate that the type was not changed."""
+        error_msg = None
+        is_valid = True
+        current_type = self.current_file.get('type', {})
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/incident_field.py]: why default is a dict? it should be a string

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -130,7 +130,6 @@ def filter_packagify_changes(modified_files, added_files, removed_files, tag='ma
                     details.get('toversion', '99.99.99')
                 ])
                 packagify_diff[uniq_identifier] = file_path
-
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py]: restore the new line

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 4]
```diff
@@ -354,9 +354,26 @@ def is_valid_required(self):
         # as can be seen in this pr: https://github.com/demisto/content/pull/5682
         required = self.current_file.get('required', False)
         if required:
-            error_msg = f'{self.file_path}: new incident fields can not be required' \
-                        f' due to a current platform limitation.'
+            error_msg = f'{self.file_path}: new incident fields can not be required.' \
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/incident_field.py]: new fields can be required if they are associated with new incident type.

File: demisto_sdk/commands/common/hook_validations/incident_field.py [hunk 5]
```diff
@@ -160,15 +160,15 @@ class IncidentFieldValidator(BaseValidator):
...
+        """Check whether the Incident Field is backward compatible or not
         """
         if not self.old_file:
             return True
 
         is_bc_broke = any(
             [
-                self.is_changed_from_version(),
+                self.is_changed_type(),
+                self.is_changed_from_version()
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/incident_field.py]:
```suggestion
                self.is_changed_from_version(),
```

File: demisto_sdk/commands/common/hook_validations/incident_field.py (hunk 4 above)
Comment (bakatzir) [reply demisto_sdk/commands/common/hook_validations/incident_field.py]: I wish, but it is not true due to a platform limitation.
https://github.com/demisto/etc/issues/22069

**Pull request merged**